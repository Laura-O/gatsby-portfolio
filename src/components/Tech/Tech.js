import React from 'react';
import styled from 'styled-components';

import * as animations from '../../layouts/shared/animations';
import * as constants from '../../layouts/shared/style-constants';

const TechItem = styled.div`
  display: flex;
  justify-content: space-around;
  width: 100%;
 
`;

const TechBox = styled.div`
  margin: 5px;
  border-radius: 5px;
  padding: 5px;
  text-align: center;
  animation: ${animations.moveInTop} 1s ease-out;

  h3 {
    transition: all 0.5s ease;
    animation: ${animations.rotateHeading} 1.2s ease-in;

    &:hover {
      text-shadow: 0.2vmin 0.2vmin 0 lighten(${constants.redPurple}, 20);
    }
  }
`;

const TechTags = styled.ul`
  margin: 0;
  padding: 0;

  li {
    list-style-type: none;
    margin: 2px;
    z-index: 0;
    text-align: center;
  }
`;

const Tech = () => {
  const frontEnd = ['JavaScript', 'Elm', 'React/Redux, Gatsby, Vue', 'HTML5', 'SCSS'];
  const backEnd = ['Node', 'Express', 'PostgreSQL'];
  const other = ['Git', 'R'];

  return (
    <TechItem>
      <h2>Skills</h2>
      <TechBox>
        <h3>Frontend</h3>
        <TechTags>{frontEnd.map(tech => <li key={tech}>{tech}</li>)}</TechTags>
      </TechBox>
      <TechBox>
        <h3>Backend</h3>
        <TechTags>{backEnd.map(tech => <li key={tech}>{tech}</li>)}</TechTags>
      </TechBox>
      <TechBox>
        <h3>Other</h3>
        <TechTags>{other.map(tech => <li key={tech}>{tech}</li>)}</TechTags>
      </TechBox>
    </TechItem>
  );
};

export default Tech;
