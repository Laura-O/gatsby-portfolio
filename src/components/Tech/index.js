import React from 'react';
import Link from 'gatsby-link';

import styles from './styles.module.scss';

const Tech = ({ stack }) => {
  const frontEnd = ['React', 'Redux', 'Gatsby', 'Vue.js', 'HTML5', 'SCSS'];
  const backEnd = ['Node', 'Express', 'PostgreSQL'];
  const other = ['Git', 'R'];

  return (
    <div className={styles.tech}>
      <h2>Skills</h2>
      <div className={styles.techbox}>
        <h1>Frontend</h1>
        <ul className={styles.tags}>
          {frontEnd.map(tech => <li key={tech}>{tech}</li>)}
        </ul>
      </div>
      <div className={styles.techbox}>
        <h1>Backend</h1>
        <ul className={styles.tags}>
          {backEnd.map(tech => <li key={tech}>{tech}</li>)}
        </ul>
      </div>
      <div className={styles.techbox}>
        <h1>Other</h1>
        <ul className={styles.tags}>
          {other.map(tech => <li key={tech}>{tech}</li>)}
        </ul>
      </div>
    </div>
  );
};

export default Tech;
