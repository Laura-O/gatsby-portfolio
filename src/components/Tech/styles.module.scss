@import '../../scss/variables.scss';

.tech {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  width: 100%;
  border-top: 3px solid $redPurple;
  z-index: inherit;
  padding: 10px 15px;
}

.techbox {
  margin: 10px;
  border-radius: 5px;
  padding: 5px;
  text-align: center;

  h1 {
    -webkit-transform: skew(-7deg) rotate(-7deg);
    transform: skew(-7deg) rotate(-7deg);
    font-size: 2em;
    text-shadow: 0.3vmin 0.3vmin 0 lighten($malachite, 40);
  }
}

.tags {
  margin: 0;
  padding: 0;

  li {
    list-style-type: none;
    font-size: 1em;
    margin: 2px;
    z-index: 0;
    text-align: center;
    animation: fadeIn 0.2s linear;
    animation-fill-mode: both;
  }

  @for $i from 1 through 10 {
    li:nth-child(#{$i}) {
      animation-delay: 0.1s * $i;
    }
  }

  @keyframes fadeIn {
    0% {
      opacity: 0;
      top: 100px;
    }
    75% {
      opacity: 0.5;
      top: 0px;
    }
    100% {
      opacity: 1;
    }
  }
}
