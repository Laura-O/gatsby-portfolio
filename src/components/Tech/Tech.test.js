import React from 'react';
import { shallow } from 'enzyme';
import Tech from './Tech';

test('rendering', () => {
  const wrapper = shallow(<Tech path="/something" />);
  expect(wrapper).toMatchSnapshot();
});
