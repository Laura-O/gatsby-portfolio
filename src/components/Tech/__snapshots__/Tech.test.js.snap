// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`rendering 1`] = `
<div>
  <h2>
    Skills
  </h2>
  <div>
    <h1>
      Frontend
    </h1>
    <ul>
      <li
        key="React"
      >
        React
      </li>
      <li
        key="Redux"
      >
        Redux
      </li>
      <li
        key="Gatsby"
      >
        Gatsby
      </li>
      <li
        key="Vue.js"
      >
        Vue.js
      </li>
      <li
        key="HTML5"
      >
        HTML5
      </li>
      <li
        key="SCSS"
      >
        SCSS
      </li>
    </ul>
  </div>
  <div>
    <h1>
      Backend
    </h1>
    <ul>
      <li
        key="Node"
      >
        Node
      </li>
      <li
        key="Express"
      >
        Express
      </li>
      <li
        key="PostgreSQL"
      >
        PostgreSQL
      </li>
    </ul>
  </div>
  <div>
    <h1>
      Other
    </h1>
    <ul>
      <li
        key="Git"
      >
        Git
      </li>
      <li
        key="R"
      >
        R
      </li>
    </ul>
  </div>
</div>
`;
